don't crash on new files because they have no git history

```
[ERROR] FileNotTrackedError: Failed to retrieve the git history for file "docs/misc/tools/minimessage-web-editor.md" because the file is not tracked by git.
    at matchFromRegex (F:\Projects\paperdocs\node_modules\.pnpm\@docusaurus+utils@3.2.1_patch_hash=yygnmtjhmebpawcus5yp7z64bi_@docusaurus+types@3.2.1_esbuild@0.20.2\node_modules\@docusaurus\utils\src\gitUtils.ts:181:11)
    at getFileCommitHash (F:\Projects\paperdocs\node_modules\.pnpm\@docusaurus+utils@3.2.1_patch_hash=yygnmtjhmebpawcus5yp7z64bi_@docusaurus+types@3.2.1_esbuild@0.20.2\node_modules\@docusaurus\utils\src\gitUtils.ts:233:17)
    at async Promise.all (index 14)
    at async cacheAuthorData (F:\Projects\paperdocs\src\util\authorUtils.ts:62:19)

```

--- a/docusaurus.config.ts
+++ b/docusaurus.config.ts
@@ -10,7 +10,7 @@ import { getFileCommitHash } from "@docusaurus/utils/src/gitUtils";
 import { AUTHOR_FALLBACK, AuthorData, commitCache, cacheAuthorData } from "./src/util/authorUtils";
 
 const preview = env.VERCEL_ENV === "preview";
-cacheAuthorData(preview);
+cacheAuthorData(preview || process.env.NODE_ENV === "development");
 
 const url = (preview && `https://${env.VERCEL_URL}`) || "https://docs.papermc.io";
 

--- a/src/util/authorUtils.ts
+++ b/src/util/authorUtils.ts
@@ -59,16 +59,17 @@ export async function cacheAuthorData(isPreview: boolean) {
   }
 
   const pagesFiles = await Globby("docs/**/*.md*");
-  const commits = await Promise.all(pagesFiles.map((f) => {
-    return getFileCommitHash(f)
-      .catch((e) => {
+  const commits = await Promise.all(
+    pagesFiles.map((f) => {
+      return getFileCommitHash(f).catch((e) => {
         if (e instanceof FileNotTrackedError) {
           return null;
         }
 
         throw e; // different error, rethrow
-      })
-  }));
+      });
+    })
+  );
 
   const commitsSet = new Set(commits.filter(Boolean).map((value) => value.commit));
   await Promise.all(Array.from(commitsSet).map(cacheUsernameFromCommit));

--- a/src/util/authorUtils.ts
+++ b/src/util/authorUtils.ts
@@ -3,7 +3,7 @@ import path from "path";
 import { Endpoints } from "@octokit/types";
 import axios from "axios";
 
-import { getFileCommitHash } from "@docusaurus/utils/src/gitUtils";
+import { getFileCommitHash, FileNotTrackedError } from "@docusaurus/utils/src/gitUtils";
 import { Globby } from "@docusaurus/utils/src/globUtils";
 
 type endpoint = Endpoints["GET /repos/{owner}/{repo}/commits/{ref}"];
@@ -59,8 +59,17 @@ export async function cacheAuthorData(isPreview: boolean) {
   }
 
   const pagesFiles = await Globby("docs/**/*.md*");
-  const commits = await Promise.all(pagesFiles.map(getFileCommitHash));
-  const commitsSet = new Set(commits.map((value) => value.commit));
+  const commits = await Promise.all(pagesFiles.map((f) => {
+    return getFileCommitHash(f)
+      .catch((e) => {
+        if (e instanceof FileNotTrackedError) {
+          return null;
+        }
 
+        throw e; // different error, rethrow
+      })
+  }));
+
+  const commitsSet = new Set(commits.filter(Boolean).map((value) => value.commit));
   await Promise.all(Array.from(commitsSet).map(cacheUsernameFromCommit));
 }

